<p>A long parameter list can indicate that a new structure should be created to wrap the numerous parameters or that the function is doing too many
things.</p>
<h2>Noncompliant Code Example</h2>
<p>With a maximum number of 4 parameters:</p>
<pre>
fun foo(p1: String, p2: String, p3: String, p4: String, p5: String) { // Noncompliant
  // ...
}
</pre>
<h2>Compliant Solution</h2>
<pre>
fun foo(p1: String, p2: String, p3: String, p4: String) {
  // ...
}
</pre>
<h2>Exceptions</h2>
<p>Methods annotated with Springâ€™s <code>@RequestMapping</code> (and related shortcut annotations, like <code>@GetRequest</code>) or
<code>@JsonCreator</code> may have a lot of parameters, encapsulation being possible. Such methods are therefore ignored.</p>

