<h2>Why is this an issue?</h2>
<p>Code annotated as deprecated should not be used since it will be removed sooner or later.</p>
<h3>Noncompliant code example</h3>
<pre>
@Deprecated("")
interface Old

class Example : Old // Noncompliant
</pre>

