<p>Code annotated as deprecated should not be used since it will be removed sooner or later.</p>
<h2>Noncompliant Code Example</h2>
<pre>
@Deprecated("")
interface Old

class Example : Old // Noncompliant
</pre>

