<p>Having two clauses in a <code>when</code> statement or two branches in an <code>if</code> chain with the same implementation is at best duplicate
code, and at worst a coding error. If the same logic is truly needed for both instances, then in an <code>if</code> chain they should be combined, or
for a <code>when</code>, duplicates should be refactored. </p>
<h2>Noncompliant Code Example</h2>
<pre>
fun s1871(x: Int) {
    when (x) {
        1 -&gt; {
            val y = x / 2
            print(y)
        }
        2 -&gt; {
            val y = x / 2
            print(y)
        }
    }
}
</pre>
<h2>Exceptions</h2>
<p>Blocks in an <code>if</code> chain that contain a single line of code are ignored, as are blocks in a <code>when</code> statement that contain a
single line of code with or without a following <code>break</code>.</p>
<pre>
if (a == 1) {
    doSomething()  //no issue, usually this is done on purpose to increase the readability
} else if (a == 2) {
    doSomethingElse()
} else {
    doSomething()
}
</pre>
<p>But this exception does not apply to <code>if</code> chains without <code>else</code>-s, or to <code>when</code>-es without <code>else</code>
clauses when all branches have the same single line of code. In case of <code>if</code> chains with <code>else</code>-s, or of <code>when</code>-es
with default clauses, rule {rule:kotlin:S3923} raises a bug. </p>
<pre>
if (a == 1) {
  doSomething()  //Noncompliant, this might have been done on purpose but probably not
} else if (a == 2) {
  doSomething()
}
</pre>

