<h2>Why is this an issue?</h2>
<p>When two functions have the same implementation, either it was a mistake - something else was intended - or the duplication was intentional, but
may be confusing to maintainers. In the latter case, the code should be refactored.</p>
<h3>Noncompliant code example</h3>
<pre>
class MyClass {
  fun calculateCode(): String {
    doTheThing()
    doOtherThing()
    return "done"
  }

  fun getStatus(): String {  // Noncompliant
    doTheThing()
    doOtherThing()
    return "done"
  }
}
</pre>
<h3>Compliant solution</h3>
<pre>
class MyClass {
  fun calculateCode(): String {
    doTheThing()
    doOtherThing()
    return "done"
  }

  fun getStatus(): String = calculateCode()
}
</pre>
<h3>Exceptions</h3>
<p>Methods with fewer than 2 statements are ignored.</p>

