<h2>Why is this an issue?</h2>
<p>An empty multi-line comment is likely to be a mistake and doesnâ€™t help to improve the readability of the code. For these reasons, it should be
removed.</p>
<h3>Noncompliant code example</h3>
<pre>
/* */

/*

 */
</pre>

