<p>An empty multi-line comment is likely to be a mistake and doesnâ€™t help to improve the readability of the code. For these reasons, it should be
removed.</p>
<h2>Noncompliant Code Example</h2>
<pre>
/* */

/*

 */
</pre>

