<p>When using POSIX classes like <code>\p{Alpha}</code> without the <code>(?U)</code> to include Unicode characters or when using hard-coded character
classes like <code>"[a-zA-Z]"</code>, letters outside of the ASCII range, such as umlauts, accented letters or letter from non-Latin languages, wonâ€™t
be matched. This may cause code to incorrectly handle input containing such letters.</p>
<p>To correctly handle non-ASCII input, it is recommended to use Unicode classes like <code>\p{IsAlphabetic}</code>. When using POSIX classes, Unicode
support should be enabled by using <code>(?U)</code> inside the regex.</p>
<h2>Noncompliant Code Example</h2>
<pre>
Regex("[a-zA-Z]")
Regex("\\p{Alpha}")
Regex("""\p{Alpha}""")
</pre>
<h2>Compliant Solution</h2>
<pre>
Regex("""\p{IsAlphabetic}""") // matches all letters from all languages
Regex("""\p{IsLatin}""") // matches latin letters, including umlauts and other non-ASCII variations
Regex("""(?U)\p{Alpha}""")
Regex("(?U)\\p{Alpha}")
</pre>

