<p>Multi-line string literals representing a big chunk of text should not be used directly in complex expressions, as it decreases the code’s
readability. It is preferred to extract the string literal and store it in a variable or a field that can be accessed in the context of the complex
expression.</p>
<p>This rule reports an issue when a multi-line string literal consists of more lines than specified in the rule’s parameter and is directly used
within a lambda literal.</p>
<h2>Noncompliant Code Example</h2>
<pre>
listOfString
    .map { str -&gt;
        str != """
        &lt;project&gt;
          &lt;modelVersion&gt;4.0.0&lt;/modelVersion&gt;
          &lt;parent&gt;
            &lt;groupId&gt;com.mycompany.app&lt;/groupId&gt;
            &lt;artifactId&gt;my-app&lt;/artifactId&gt;
            &lt;version&gt;1&lt;/version&gt;
          &lt;/parent&gt;

          &lt;groupId&gt;com.mycompany.app&lt;/groupId&gt;
          &lt;artifactId&gt;my-module&lt;/artifactId&gt;
          &lt;version&gt;1&lt;/version&gt;
        &lt;/project&gt;
        """
    }
</pre>
<h2>Compliant Solution</h2>
<pre>
val myTextBlock = """
    &lt;project&gt;
      &lt;modelVersion&gt;4.0.0&lt;/modelVersion&gt;
      &lt;parent&gt;
        &lt;groupId&gt;com.mycompany.app&lt;/groupId&gt;
        &lt;artifactId&gt;my-app&lt;/artifactId&gt;
        &lt;version&gt;1&lt;/version&gt;
      &lt;/parent&gt;

      &lt;groupId&gt;com.mycompany.app&lt;/groupId&gt;
      &lt;artifactId&gt;my-module&lt;/artifactId&gt;
      &lt;version&gt;1&lt;/version&gt;
    &lt;/project&gt;
    """

listOfString.map { str -&gt; str != myTextBlock }
</pre>
<h2>See</h2>
<ul>
  <li> <a href="https://kotlinlang.org/spec/expressions.html#string-interpolation-expressions">8.2 String interpolation expressions</a> in the Kotlin
  language specification </li>
</ul>

